### PR TITLE
M2P-450 Remove legacy unit test flow from Circle CI

### PR DESCRIPTION
**Pull request by dumega**
# Description

- Remove jobs like unit-php74-magento24 and add jobs like integration-magento-php74-magento24 instead

- Remove function skipTestInUnitTestsFlow

- Update ReadMe https://github.com/BoltApp/bolt-magento2/blob/master/Test/Unit/README.md

- Check if we calculate test coverage correctly

Fixes: https://boltpay.atlassian.net/browse/M2P-450

#changelog Remove legacy unit test flow from Circle CI

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 
- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Jira ticket link and provided a changelog message.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -288,7 +288,7 @@ workflows:
   build:
     when: << pipeline.parameters.run_default_workflow >>
     jobs:
-      - unit-php74-magento24
+      - integration-php74-magento24
       - unit-php72-magento23
       - unit-php71-magento22
       - unit-php70-magento22
```
Comment (danac-gs) [inline .circleci/config.yml:294]: we should also remove PHP7.0 tests, as it is no longer supported by Magento
PHP7.1 and PHP7.2 are no longer recommended as well but may keep them for now

Comment (dumega) [reply .circleci/config.yml:294]: @danac-gs yes, this PR is still WIP, I will let you know once it is ready. Thanks

Comment (danac-gs) [reply .circleci/config.yml]: @dumega for WIP you can create a draft PR, or convert it to draft as in the picture
image

**Comment by dumega** [comment]
@adentong 
1. Can you update the M2 verion in the image boltdev/m2-installed-plugin-ci-php71:2.2.0-v1 from 2.2.0 to 2.2.8? Cause M2 v2.2.0 does not have the class Magento\Sales\Model\OrderIncrementIdChecker which is required in our plugin (https://github.com/BoltApp/bolt-magento2/blob/ning-remove-legacy-unit-test/Helper/Order.php#L49)


2. Also can you update the M2 verion in the image boltdev/m2-installed-plugin-ci-php74:2.4.0-v1 from 2.4.0 to 2.4.2? Cause for M2 2.4.0 has a bug with integration test https://github.com/dotmailer/dotmailer-magento2-extension/issues/566, and this bug results in the failure https://app.circleci.com/pipelines/github/BoltApp/bolt-magento2/8431/workflows/91dc7f3b-00bf-4ae8-9cf5-9cc96567fe87/jobs/32699

**Comment by adentong** [comment]
> @adentong
> 
> 1. Can you update the M2 verion in the image boltdev/m2-installed-plugin-ci-php71:2.2.0-v1 from 2.2.0 to 2.2.8? Cause M2 v2.2.0 does not have the class Magento\Sales\Model\OrderIncrementIdChecker which is required in our plugin (https://github.com/BoltApp/bolt-magento2/blob/ning-remove-legacy-unit-test/Helper/Order.php#L49)
> 2. Also can you update the M2 verion in the image boltdev/m2-installed-plugin-ci-php74:2.4.0-v1 from 2.4.0 to 2.4.2? Cause for M2 2.4.0 has a bug with integration test [dotmailer/dotmailer-magento2-extension#566](https://github.com/dotmailer/dotmailer-magento2-extension/issues/566), and this bug results in the failure https://app.circleci.com/pipelines/github/BoltApp/bolt-magento2/8431/workflows/91dc7f3b-00bf-4ae8-9cf5-9cc96567fe87/jobs/32699

Will do tomorrow.

**Comment by adentong** [comment]
@dumega I got it working. But there are some actual test failures. Once those are fixed we're good to go.

File: .circleci/config.yml [hunk 2]
```diff
@@ -254,7 +230,16 @@ jobs:
...
+      - image: selenium/standalone-chrome:3.141.59-20200326
+    resource_class: large
+    environment:
+      MAGENTO_DIR: magento
     steps:
+      - m2-php-test:
+          cache_key_separator: php72mag23installed
+          m2_version: "2.3.0"
+          php_version: "php72"
+          is_integration: true
```
Comment (adentong) [inline .circleci/config.yml:242]: maybe you want is_magento_integration: true? same below

**Comment by adentong** [comment]
Also it says "assertRegExp() is deprecated and will be removed in PHPUnit 10. Refactor your code to use assertMatchesRegularExpression() instead." Can you change to assertMatchesRegularExpression?

File: .circleci/config.yml [hunk 3]
```diff
@@ -254,7 +230,16 @@ jobs:
...
+      - image: selenium/standalone-chrome:3.141.59-20200326
+    resource_class: large
+    environment:
+      MAGENTO_DIR: magento
     steps:
+      - m2-php-test:
+          cache_key_separator: php72mag23installed
+          m2_version: "2.3.0"
+          php_version: "php72"
+          is_magento_integration: true
```
Comment (adentong) [inline .circleci/config.yml:242]: Not sure why you added this

File: .circleci/config.yml [hunk 4]
```diff
@@ -267,7 +252,16 @@ jobs:
...
```
Comment (adentong) [inline .circleci/config.yml:264]: Same here

File: .circleci/config.yml (hunk 2 above)
Comment (adentong) [inline .circleci/config.yml:242]: Not sure why you added this

File: .circleci/config.yml [hunk 5]
```diff
@@ -267,7 +252,16 @@ jobs:
...
+      - image: selenium/standalone-chrome:3.141.59-20200326
+    resource_class: large
+    environment:
+      MAGENTO_DIR: magento
     steps:
+      - m2-php-test:
+          cache_key_separator: php72mag23installed
+          m2_version: "2.3.0"
+          php_version: "php72"
+          is_integration: true
```
Comment (adentong) [inline .circleci/config.yml:264]: Same here

**Comment by adentong** [comment]
Fixed the php74 test warning ^

**Review by danac-gs — APPROVED** [review]

**Review by adentong — APPROVED** [review]
👍

**Pull request merged**